src/Symfony/Component/HttpFoundation/BinaryFileResponse.php

```diff
@@ -157,6 +157,20 @@ public function setContentDisposition($disposition, $filename = '', $filenameFal
             $filename = $this->file->getFilename();
         }
 
+        if ('' === $filenameFallback && (!preg_match('/^[\x20-\x7e]*$/', $filename) || false !== strpos($filename, '%'))) {
```

Comment:
the if block should be moved in ResponseHeaderBag::makeDisposition IMO, isn't it?